internal/provider/data_source_public_key.go

```diff
@@ -51,43 +59,73 @@ func dataSourcePublicKey() *schema.Resource {
 	}
 }
 
-func dataSourcePublicKeyRead(d *schema.ResourceData, _ interface{}) error {
-	// Read private key
-	var bytes []byte
-	if v, ok := d.GetOk("private_key_pem"); ok {
-		bytes = []byte(v.(string))
-	} else {
-		return fmt.Errorf("invalid private key %#v", v)
+func readDataSourcePublicKey(d *schema.ResourceData, _ interface{}) error {
+	prvKeyPEMArg, prvKeyPEMArgOK := d.GetOk("private_key_pem")
+	prvKeyOpenSSHPEMArg, prvKeyOpenSSHPEMArgOK := d.GetOk("private_key_openssh")
+
+	// Confirm that not both ways to provide a private key were used at the same time
+	if prvKeyPEMArgOK && prvKeyOpenSSHPEMArgOK {
+		return fmt.Errorf("either provide private key via `private_key_pem` or `private_key_openssh`, not both")
 	}
```

Comment:
The [`Schema.ExactlyOneOf` field](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema#Schema.ExactlyOneOf) can be set on both the attributes to validate this automatically during configuration validation.

```suggestion
```